lib/github.com/mpllang/mpllib/GIF.sml

```diff
@@ -298,17 +298,21 @@ struct
       ; nextCode := Util.boundPow2 numColors + 2
       )
 
-    fun insert (code, idx) ({nextCode, numColors, table}: t) =
-      if !nextCode = 4095 then
-        false (* GIF limits the maximum code number to 4095 *)
-      else
-        ( Array.update (table, code, (idx, !nextCode) :: Array.sub (table, code))
-        ; nextCode := !nextCode + 1
-        ; true
-        )
+    fun insert (code, idx) ({nextCode, numColors=_, table}: t) =
+      let
+        (* GIF limits the maximum code number to 4095 *)
+        val ret = not (!nextCode <> 4095)
+      in
+        if ret then
+          ( Array.update (table, code, (idx, !nextCode) :: Array.sub (table, code))
+          ; nextCode := !nextCode + 1
+          )
+        else ();
+        ret
+      end
```

Comment:
I don't see anything wrong with the first version? It's much easier to read before the edit...